### PR TITLE
gh-105481: opcode.h is no longer generated during the build

### PR DESCRIPTION
**Pull request by iritkatriel**
Python's operators don't change very often. This doesn't need to be auto-generated.

* Issue: gh-105481

### CODE REVIEW

File: Makefile.pre.in [hunk 1]
```diff
@@ -1427,13 +1427,11 @@ regen-ast:
 
 .PHONY: regen-opcode
 regen-opcode:
-	# Regenerate Include/opcode.h from Lib/opcode.py
+	# Regenerate Include/internal/pycode_opcode.h from Lib/opcode.py
```
Comment (iritkatriel) [inline Makefile.pre.in]:
```suggestion
	# Regenerate Include/internal/pycore_opcode.h from Lib/opcode.py
```

File: Include/opcode.h [hunk 2]
```diff
@@ -36,6 +34,7 @@ extern "C" {
 #define NB_INPLACE_TRUE_DIVIDE                  24
 #define NB_INPLACE_XOR                          25
 
+#define NB_OPARG_LAST                           25
```
Comment (gvanrossum) [inline Include/opcode.h:37]: I dislike marking the "last" of anything, you end up doing `+ 1` in various places. But I realize that it's a convention around opcode definitions and the like.

**Review by gvanrossum — APPROVED** [review]
LG, though I'm unsure of the benefit here. We now have two places where the list of NB_ constants occurs, once in opcode.h, once in _opcode.c, and they must be kept in sync manually.

**Comment by iritkatriel** [comment]
There are a number of lists/enumerations that need to exist in parallel python and C. I think to avoid the duplication in the code we should design something generic that spawns the #definitions in C and a function that builds the python dict to map the name to the number. Something like argument clinic, but for enumerations.  I don't like having a special script for each case.

**Pull request merged**